### PR TITLE
Normal smoothing regularization like unisurf

### PR DESCRIPTION
**Pull request by joebartusek**
Unisurf features random uniform 3D perturbation for calculating smooth normals loss

--smooth-eps argument determines magnitude of the perturbation; if 0, this regularization is not used

### CODE REVIEW

File: runner.py [hunk 1]
```diff
@@ -431,11 +435,17 @@ def train(model, cam, labels, opt, args, light=None, sched=None):
...
     # TODO does it matter to normalize the normal or not?
     if args.smooth_normals > 0:
-      # TODO maybe lower dimensionality of n?
-      delta_n = torch.autograd.grad(
-        inputs=pts, outputs=F.normalize(n,dim=-1), create_graph=True,
-        grad_outputs=torch.ones_like(n),
-      )[0]
+      if args.smooth_eps > 0:
+        # epsilon-perturbation implementation from unisurf
+        perturbation = F.normalize(torch.rand(3, device=device), dim=-1) * args.smooth_eps
```
Comment (JulianKnodt) [inline runner.py]: Just curious, will this have some bias because it's not uniform on `[-1,1]`, but `[0,1]`?

File: runner.py [hunk 2]
```diff
@@ -431,11 +435,17 @@ def train(model, cam, labels, opt, args, light=None, sched=None):
...
+        perturbation = perturbation.expand(pts.size())
```
Comment (JulianKnodt) [inline runner.py]: super tiny nit, but would prefer shape instead of size

**Review by JulianKnodt — APPROVED** [review]
